### ISSUE TITLE
Exception in the Eclipse Code Formatter

### ISSUE THREAD

**Issue by Lamo152** [krasa/EclipseCodeFormatter#143]
``` -1
java.lang.ArrayIndexOutOfBoundsException: -1
	at java.util.ArrayList.elementData(ArrayList.java:418)
	at java.util.ArrayList.get(ArrayList.java:431)
	at org.eclipse.jdt.internal.formatter.TokenManager.get(Unknown Source)
	at org.eclipse.jdt.internal.formatter.TokenManager.lastTokenIn(Unknown Source)
	at org.eclipse.jdt.internal.formatter.linewrap.WrapPreparator.preVisit2(Unknown Source)
	at org.eclipse.jdt.core.dom.ASTNode.accept(Unknown Source)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChildren(Unknown Source)
	at org.eclipse.jdt.core.dom.FieldDeclaration.accept0(Unknown Source)
	at org.eclipse.jdt.core.dom.ASTNode.accept(Unknown Source)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChildren(Unknown Source)
	at org.eclipse.jdt.core.dom.TypeDeclaration.accept0(Unknown Source)
	at org.eclipse.jdt.core.dom.ASTNode.accept(Unknown Source)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChildren(Unknown Source)
	at org.eclipse.jdt.core.dom.CompilationUnit.accept0(Unknown Source)
	at org.eclipse.jdt.core.dom.ASTNode.accept(Unknown Source)
	at org.eclipse.jdt.internal.formatter.DefaultCodeFormatter.prepareWraps(Unknown Source)
	at org.eclipse.jdt.internal.formatter.DefaultCodeFormatter.prepareFormattedCode(Unknown Source)
	at org.eclipse.jdt.internal.formatter.DefaultCodeFormatter.format(Unknown Source)
	at org.eclipse.jdt.internal.formatter.DefaultCodeFormatter.format(Unknown Source)
	at krasa.formatter.adapter.EclipseJavaFormatterAdapter.format(EclipseJavaFormatterAdapter.java:82)
	at krasa.formatter.eclipse.JavaCodeFormatterFacade.format(JavaCodeFormatterFacade.java:55)
	at krasa.formatter.plugin.EclipseCodeFormatter.reformat(EclipseCodeFormatter.java:112)
	at krasa.formatter.plugin.EclipseCodeFormatter.formatWhenEditorIsOpen(EclipseCodeFormatter.java:90)
	at krasa.formatter.plugin.EclipseCodeFormatter.format(EclipseCodeFormatter.java:55)
	at krasa.formatter.plugin.EclipseCodeStyleManager.formatWithEclipse(EclipseCodeStyleManager.java:224)
	at krasa.formatter.plugin.EclipseCodeStyleManager.format(EclipseCodeStyleManager.java:115)
	at krasa.formatter.plugin.EclipseCodeStyleManager.reformatText(EclipseCodeStyleManager.java:78)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at krasa.formatter.plugin.CodeStyleManagerDelegator.intercept(CodeStyleManagerDelegator.java:42)
	at com.intellij.psi.impl.source.codeStyle.CodeStyleManagerImpl$$EnhancerByMockitoWithCGLIB$$7d449777.reformatText(<generated>)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$prepareTask$0(ReformatCodeProcessor.java:129)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$null$0(AbstractLayoutCodeProcessor.java:239)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1023)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$preprocessFile$1(AbstractLayoutCodeProcessor.java:239)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$runProcessFile$3(AbstractLayoutCodeProcessor.java:342)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$null$6(AbstractLayoutCodeProcessor.java:422)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:149)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:109)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:99)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:85)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$null$7(AbstractLayoutCodeProcessor.java:420)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:303)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:410)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:399)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:827)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:655)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82) ```

**Comment by krasa** [comment]
Can you share the code which reproduces it?

**Comment by krasa** [comment]
Fixed by #144
